src/components/tabs/Tab.vue

```diff
@@ -1,18 +1,25 @@
 <template>
-  <div class="tabs-pane" :id="id">
+  <div class="tabs-pane" :id="hash">
     <slot></slot>
   </div>
 </template>
 
 <script>
+  import {createUniqueId} from '../../utils'
+
   export default {
     props: {
-      tabId: String
+      id: {
+        type: String
+      },
+      name: {
+        type: String,
+        required: true
+      }
     },
-
-    data() {
-      return {
-        id: this.tabId
+    computed: {
+      hash() {
+        return this.id ? this.id : createUniqueId(`tab`)
```

Comment:
I think it's actually a really good idea to embed default id generation in the `Tab.vue`, it's used the same way in other places in this project. One suggestion, I think it would be nicer to use this pattern


    id: {
      type: String,
      default: () => createUniqueId('tab')
    },

Reply:
Then we also won't need the extra internal variable `hash` which (the name) doesn't sound convincing to me :)

Reply:
Please check my last commit a6357b226cf1e1da626b13ff122334c501ab6344